### PR TITLE
Gekidou Sentry Installation

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary

This is the first PR related to the Sentry feature.

#### ACHIEVED
In here, the following have been tested: 

1. successful sourcemap upload for Android
2. successful sourcemap upload for iOS
3. the error event shows the line number for Android
4. the error event shows the line number for iOS
5. integration with the react-native-navigation lib for Performance monitoring

For iOS extensions, Sentry do catch the errors as describe on [this issue](https://github.com/getsentry/sentry-cocoa/issues/1656#issuecomment-1263336691)

#### Screenshots
Sentry - Contexts
sentry - pinpoint line number
Sentry - pinpoint line number - A
[Sentry - whole page.pdf](https://github.com/mattermost/mattermost-mobile/files/9989887/Sentry.-.whole.page.pdf)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/screens/home/channel_list/channel_list.tsx
Comment (larkox) [inline app/screens/home/channel_list/channel_list.tsx:121]: Isn't this the same as just `servers.reduce((a, b) => (b.lastActiveAt > a.lastActiveAt ? b : a))` (without the code around it)

File: app/utils/sentry.ts
Comment (larkox) [inline app/utils/sentry.ts:184]: Use a function from `app/database/queries`?

File: app/utils/sentry.ts
Comment (larkox) [inline app/utils/sentry.ts:194]: Same comment as the previous one.

**Review by larkox — CHANGES_REQUESTED** [review]
The biggest change requested is regarding the patch. A quick look at it, it seems it is adding way too much more than probably intended. Can you check?

Apart of that, I also wonder if this should have some security review just to be on the safe side regarding not storing information we shouldn't.

File: app/screens/home/channel_list/channel_list.tsx
Comment (avinashlng1080) [reply app/screens/home/channel_list/channel_list.tsx]: Corrected

File: app/utils/sentry.ts
Comment (avinashlng1080) [reply app/utils/sentry.ts]: Corrected

**Comment by avinashlng1080** [comment]
@larkox - the patch file has been removed

**Comment by avinashlng1080** [comment]
@enahum  and @larkox  - I uploaded the screenshots.  Please let me know if there's any correction to be made to this PR.

**Comment by enahum** [comment]
@avinashlng1080 the third screenshot looks like is pointing to the `Sentry.captureException` function as the source that caused the exception, this is throwing me off, also the second screenshot seems to be highlighting the wrong line.

What is the purpose of this
image

Also, can we include the navigation stack?

File: .circleci/config.yml
Comment (enahum) [inline .circleci/config.yml:569]: this context should be removed

File: app/screens/home/channel_list/channel_list.tsx
Comment (enahum) [inline app/screens/home/channel_list/channel_list.tsx:143]: this does not look like the right place for this, and if it is then you have access to the `serverUrl` with `useServerUrl` removing the need for the activeServerUrlObserver and subscribeActiveServers.

maybe this makes more sense to have at `home/index.tsx` ?

File: app/utils/sentry.ts
Comment (enahum) [inline app/utils/sentry.ts:153]: do we care?

File: app/utils/sentry.ts
Comment (enahum) [inline app/utils/sentry.ts:180]: do we care?

File: app/utils/sentry.ts
Comment (enahum) [inline app/utils/sentry.ts:192]: do we care?

this questions are to trigger a discussion/conversation

**Review by enahum — CHANGES_REQUESTED** [review]
Also, can you share data being captured from the Navigation library around perf?

**Comment by enahum** [comment]
@avinashlng1080 why are there multiple Channel and Thread screens in that report?

**Comment by avinashlng1080** [comment]
> @avinashlng1080 why are there multiple Channel and Thread screens in that report?
The `transaction/screen` column is repeating but the operation column is unique.

Sentry - navigation - transacitons

**Comment by enahum** [comment]
> > @avinashlng1080 why are there multiple Channel and Thread screens in that report?
> > The `transaction/screen` column is repeating but the operation column is unique.
> 
> Sentry - navigation - transacitons

can it be aggregated? I guess that is fine for now, we can fine tune if needed at a later time

**Review by larkox — APPROVED** [review]
Nothing from my side, apart of what enahum already said.

File: app/utils/sentry.ts
Comment (avinashlng1080) [reply app/utils/sentry.ts:192]: Thanks @enahum .  On v1, these information are being collected and sent to Sentry.  Maybe to be used for deeper investigation, I'm not sure.  I tried to replicate what was in place.

If they are redundant and that from v1 experience, we can remove them.  Should the need arise in the future to bring in more `contexts` as to what channel/team the user had these problems, then we can add those back.

Comment (enahum) [reply app/utils/sentry.ts:192]: Yes let's remove the team and channel context and address the other comment so we csn merge this.

Once merged ping me to give you instructions before cutting the build

File: .circleci/config.yml
Comment (avinashlng1080) [reply .circleci/config.yml]: removed

File: app/screens/home/channel_list/channel_list.tsx
Comment (avinashlng1080) [reply app/screens/home/channel_list/channel_list.tsx]: moved to home.index

File: app/utils/sentry.ts
Comment (avinashlng1080) [reply app/utils/sentry.ts]: removed team and channel

File: app/utils/sentry.ts
Comment (avinashlng1080) [reply app/utils/sentry.ts:151]: roles and locale can come in useful.
e.g. timezone change when we got in Barbados last time.

**Review by enahum — APPROVED** [review]

**Pull request merged**
